File: src/main/java/hu/frontrider/arcana/capabilities/ICreatureEnchant.java
package hu.frontrider.arcana.capabilities;

import hu.frontrider.arcana.creatureenchant.backend.CreatureEnchant;
import hu.frontrider.arcana.creatureenchant.backend.EnchantingBaseCircle;

import java.util.Map;


public interface ICreatureEnchant {

    boolean hasEnchant();

    boolean hasEnchant(CreatureEnchant enchant);

    int getLevel(CreatureEnchant enchant);

    Map<CreatureEnchant, CreatureEnchantCapability.CreatureEnchantContainer> getStore();

    void setStore(Map<CreatureEnchant, CreatureEnchantCapability.CreatureEnchantContainer> store);

    void putEnchant(CreatureEnchantCapability.CreatureEnchantContainer creatureEnchantContainer);

    EnchantingBaseCircle getCircle();
    void setCircle(EnchantingBaseCircle enchantingBaseCircle);
}

File: src/main/java/hu/frontrider/arcana/capabilities/ICreatureEnchant.kt
package hu.frontrider.arcana.capabilities

import hu.frontrider.arcana.creatureenchant.backend.CreatureEnchant
import hu.frontrider.arcana.creatureenchant.backend.EnchantingBaseCircle


interface ICreatureEnchant {

    var store: MutableMap<CreatureEnchant, CreatureEnchantCapability.CreatureEnchantContainer>

    var circle: EnchantingBaseCircle

    fun hasEnchant(): Boolean

    fun hasEnchant(enchant: CreatureEnchant): Boolean

    fun getLevel(enchant: CreatureEnchant): Int

    fun setEnabledStatus(store: MutableMap<CreatureEnchant, Boolean>)

    fun setEnabledStatus(enchant: CreatureEnchant, status: Boolean)

    fun getEnabledStatus(): MutableMap<CreatureEnchant, Boolean>

    fun putEnchant(creatureEnchantContainer: CreatureEnchantCapability.CreatureEnchantContainer)
}
